src/Data/List/Relation/Binary/Sublist/Setoid.agda

```diff
@@ -79,3 +79,80 @@ open HeterogeneousProperties.Antisymmetry {R = _≈_} {S = _≈_} (λ x≈y _ 
 ⊆-poset = record
   { isPartialOrder = ⊆-isPartialOrder
   }
+
+------------------------------------------------------------------------
+-- Weak pushout
+--
+-- The category _⊆_ does not have proper pushouts.  For instance consider:
+--
+--   τᵤ : [] ⊆ (u ∷ [])
+--   τᵥ : [] ⊆ (v ∷ [])
+--
+-- Then, there are two unrelated upper bounds (u ∷ v ∷ []) and (v ∷ u ∷ []),
+-- since _⊆_ does not include permutations.
+--
+-- Even though there are no unique least upper bounds, we can merge two
+-- extensions of a list, producing a minimial superlist of both.
+--
+-- For the example, the left-biased merge would produce the pair:
+--
+--   τᵤ′ : (u ∷ []) ⊆ (u ∷ v ∷ [])
+--   τᵥ′ : (v ∷ []) ⊆ (u ∷ v ∷ [])
+--
+-- We call such a pair a raw pushout.  It is then a weak pushout if the
+-- resulting square commutes, i.e.:
+--
+--   ⊆-trans τᵤ τᵤ′ ~ ⊆-trans τᵥ τᵥ′
+--
+-- This requires a notion of equality _~_ on sublist morphisms.
+--
+-- Further, commutation requires a similar commutation property
+-- for the underlying equality _≈_, namely
+--
+--   trans x≈y (sym x≈y) == trans x≈z (sym x≈z)
+--
+-- for some notion of equality _==_ for equality proofs _≈_.
+-- Such a property is given e.g. if _≈_ is proof irrelevant
+-- or forms a groupoid.
+
+private
+  variable
+    x y z : A
+    xs ys zs us vs : List A
+    τ τ′ τ₁ τ₂ σ σ′ : xs ⊆ ys
+
+record RawPushout (τ : xs ⊆ ys) (σ : xs ⊆ zs) : Set (c ⊔ ℓ) where
+  constructor rawPushout
+  field
+    {upperBound} : List A
+    leg₁         : ys ⊆ upperBound
+    leg₂         : zs ⊆ upperBound
+
+open RawPushout
+
+------------------------------------------------------------------------
+-- Extending corners of a raw pushout square
+
+-- Extending the right upper corner.
+
+_∷ʳ₁_ : ∀ y → RawPushout τ σ → RawPushout (y ∷ʳ τ) σ
+y ∷ʳ₁ rpo = record { leg₁ = refl ∷ leg₁ rpo ; leg₂ = y ∷ʳ leg₂ rpo }
```

Comment:
Please could layout the records as specified by the [style guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md)

Reply:
Ok.
(Maybe except the very last, which is very short.)